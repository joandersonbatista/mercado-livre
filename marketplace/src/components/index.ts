export { Header } from './header/header';
export { IconButton } from './icon-button/icon-button';
export { Button } from './button/button';
export { Breadcrumbs } from './breadcrumbs/breadcrumbs';
export { BreadcrumbsSkeleton } from './breadcrumbs/breadcrumbs-skeleton';
export { Skeleton, type SkeletonProps } from './skeleton/skeleton';
export { LanguageSwitcher } from './language-switcher/language-switcher';
export { ServerError } from './server-error/server-error';
