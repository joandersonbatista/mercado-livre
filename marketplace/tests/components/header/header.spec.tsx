import { headers } from 'next/headers';

import { Header } from '@/components';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';

jest.mock('next/headers', () => ({
  headers: jest.fn(),
}));

describe('Header Component', () => {
  beforeEach(() => {
    (headers as jest.Mock).mockReset();
  });

  it('should set the search input with the provided searchParams', async () => {
    const mockHeader = new Map();

    mockHeader.set('searchParams', 'search=teste');

    (headers as jest.Mock).mockResolvedValueOnce(mockHeader);

    const component = await Header();
    render(component);

    expect(screen.getByTestId('input-search-items')).toHaveValue('teste');
  });

  it('should update input value onChange', async () => {
    (headers as jest.Mock).mockResolvedValueOnce(new Map());

    const component = await Header();
    render(component);

    const input = screen.getByTestId('input-search-items');
    fireEvent.change(input, { target: { value: 'new search term' } });

    await waitFor(() => {
      expect(input).toHaveValue('new search term');
    });
  });

  it('should match the snapshot', async () => {
    const mockHeader = new Map();
    mockHeader.set('searchParams', 'search=playstation');
    (headers as jest.Mock).mockResolvedValueOnce(mockHeader);

    const component = await Header();
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });
});
