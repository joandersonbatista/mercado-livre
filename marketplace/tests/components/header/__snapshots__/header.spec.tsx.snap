// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<div>
  <header
    class="nav-header"
  >
    <div>
      <a
        href="/"
      >
        <img
          alt="logoMarkAlt"
          data-nimg="1"
          decoding="async"
          height="34"
          loading="lazy"
          src="/logo.svg"
          style="color: transparent;"
          width="60"
        />
      </a>
      <form
        action="/items"
        method="GET"
        role="search"
      >
        <label
          class="sr-only"
        >
          searchFormLabel
        </label>
        <input
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          data-testid="input-search-items"
          maxlength="80"
          name="search"
          placeholder="searchPlaceHolder"
          spellcheck="false"
          type="text"
          value="playstation"
        />
        <button
          class="icon-button "
          data-testid="submit-search-items"
          type="submit"
        >
          <img
            alt="searchIconAlt"
            aria-label="searchIconAriaLabel"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/icons/search.svg"
            style="color: transparent;"
            width="34"
          />
        </button>
      </form>
      <label>
        <p
          class="sr-only"
        >
          Change language
        </p>
        <select
          class="select-language-content"
        >
          <option
            value="br"
          >
            ðŸ‡§ðŸ‡·
             
            Portuguese
          </option>
          <option
            selected=""
            value="en"
          >
            ðŸ‡ºðŸ‡¸
             
            English
          </option>
          <option
            value="es"
          >
            ðŸ‡¨ðŸ‡´
             
            Spanish
          </option>
        </select>
      </label>
    </div>
  </header>
</div>
`;
